lib/srv/db/postgres/engine.go

```diff
@@ -574,6 +574,22 @@ func (e *Engine) handleCancelRequest(ctx context.Context, sessionCtx *common.Ses
 	return nil
 }
 
+// teardownUser chooses and call the auto provisioner method used to drop
+// auto-users.
+func (e *Engine) teardownUser(ctx context.Context, sessionCtx *common.Session) error {
```

Comment:
should this be moved to `common.UserProvisioner`? Seems like common logic to me.